File: src/com/hkb48/keepdo/Reminder.java
package com.hkb48.keepdo;

import java.io.Serializable;

@SuppressWarnings("serial")
public class Reminder implements Serializable {
    private boolean enabled;
    private int hourOfDay;
    private int minute;

    public Reminder() {
        this.enabled = false;
        this.hourOfDay = 0;
        this.minute = 0;
    }

    public Reminder(boolean enabled, int hourOfDay, int minute) {
        this.enabled = enabled;
        this.hourOfDay = hourOfDay;
        this.minute = minute;
    }

    boolean getEnabled() {
        return enabled;
    }

    int getHourOfDay() {
        return hourOfDay;
    }

    int getMinute() {
        return minute;
    }

    void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    void setHourOfDay(int hourOfDay) {
        this.hourOfDay = hourOfDay;
    }

    void setMinute(int minute) {
        this.minute = minute;
    }
}

File: app/src/main/java/com/hkb48/keepdo/Reminder.kt
package com.hkb48.keepdo

import java.io.Serializable
import java.util.*

class Reminder : Serializable {
    var enabled: Boolean
    var hourOfDay = 0
    var minute = 0

    constructor() {
        enabled = false
        hourOfDay = 0
        minute = 0
    }

    constructor(enabled: Boolean, timeInMillis: Long) {
        this.enabled = enabled
        this.timeInMillis = timeInMillis
    }

    var timeInMillis: Long
        get() {
            val time = Calendar.getInstance()
            time[Calendar.HOUR_OF_DAY] = hourOfDay
            time[Calendar.MINUTE] = minute
            return time.timeInMillis
        }
        private set(milliseconds) {
            val time = Calendar.getInstance()
            time.timeInMillis = milliseconds
            hourOfDay = time[Calendar.HOUR_OF_DAY]
            minute = time[Calendar.MINUTE]
        }
}